packages/instant-apps-components/src/components/instant-apps-app-guide/instant-apps-app-guide.tsx

```diff
@@ -0,0 +1,129 @@
+import { Component, Host, h, Prop, State } from '@stencil/core';
+import { Element, HostElement, Watch } from '@stencil/core/internal';
+import AppGuideViewModel from './AppGuide/AppGuideViewModel';
+import { AppGuidePage, AppGuideRenderType } from './AppGuide/interfaces/interfaces';
+import AppGuide_T9n from '../../assets/t9n/instant-apps-app-guide/resources.json';
+import { getMessages } from '../../utils/locale';
+import { ArrowType } from '@esri/calcite-components';
+
+const CSS = {
+  contentList: 'instant-apps-app-guide__content-list',
+  contentListItem: 'instant-apps-app-guide__content-list--item',
+}
+
+/**
+ * The `instant-apps-app-guide` is a component containing a page(s) that describes features of a tool or Instant App.
+ *
+ * Data for the pages is passed as an array of `AppGuidePage` objects that each have the following properties:
+ * - `title`: The title of the page; this will be displayed in the header if the `header` prop is set to true or the `header` attribute is present
+ * - `content`: An array of strings that represent the content items on the page
+ * - `type`: How the content items should be rendered. The default is 'paragraphs', but 'list' is also available for rendering as a numbered list.
+ */
+
+@Component({
+  tag: 'instant-apps-app-guide',
+  styleUrl: 'instant-apps-app-guide.scss',
+  shadow: true,
+})
+export class InstantAppsAppGuide {
+  @Element()
+  el: HTMLInstantAppsAppGuideElement;
+
+  /**
+   * Show a header with the title of the current page
+   */
+  @Prop()
+  header: boolean;
+
+  /**
+   * A collection of AppGuidePage objects that represent the content of the component
+   */
+  @Prop()
+  data: AppGuidePage[];
+
+  @Watch('data')
+  watchPropHandler(newValue: AppGuidePage[]) {
+    this.viewModel.pages = newValue;
+    this._updateHeaderText();
+  }
+
+  @State() messages: typeof AppGuide_T9n;
+  @State() headerText: string;
+
+  private viewModel: AppGuideViewModel = new AppGuideViewModel();
```

Comment:
since your private method names are prefixed with `_`, let's do that for the private variables too

Reply:
👍